### PR TITLE
ENH: Add GeoSeries.from_geojson() constructor (GH-3309)

### LINKED ISSUE #3309: ENH: GeoSeries.from_json() to parse GeoJSON strings in array-like structure, analogous to GeoSeries.from_wkt()
#### Is your feature request related to a problem?

Snowflake `GEOGRAPHY` objects are returned via the Python connector as GeoJSON strings. It's annoying to manually parse these fields, or to remember to use `st_aswkb`/`st_aswkt` when querying.

#### Describe the solution you'd like

A `GeoSeries.from_json` class method / alternate constructor, with behavior and concept analogous to `GeoSeries.from_wkt`, for reading GeoJSON strings.

#### API breaking implications

None that I'm aware of.

#### Describe alternatives you've considered

Boilerplate code like this:

```python
import json
import pandas as pd
import geopandas as gpd
from shapely.geometry import shape

def geoseries_from_json(xs: pd.Series, crs=None):
    return gpd.GeoSeries(xs.map(lambda x: shape(json.loads(x))), crs=crs)
```

### PR DESCRIPTION
**Pull request by Shikhar-Kesharwani**
- [x] closes #3309
- [x] tests added / passed
- [x] passes `pre-commit`
- [ ] entry in `CHANGELOG.md`

This PR adds an alternate constructor `GeoSeries.from_geojson()` and underlying `GeometryArray.from_geojson()` to parse GeoJSON geometry strings directly into a `GeoSeries` (analogous to `from_wkt` and `from_wkb`).

### Summary of Changes
- Implements `GeoSeries.from_geojson(data, index=None, crs=None, on_invalid='raise', **kwargs)`.
- Implements `from_geojson` in `geopandas/array.py`.
- Supports GeoJSON Geometry and Feature strings, list-like / Series inputs, missing sentinels (`None`, `NaN`, `pd.NA`), CRS propagation, and `on_invalid` options (`raise`, `warn`, `ignore`).
- Added comprehensive unit tests in `geopandas/tests/test_geoseries.py`.

---
*AI Assistance Disclosure: An AI coding assistant was used for docstring formatting and test parameterization. All implementation logic, geometry parsing, and edge-case testing were manually authored, debugged, and verified against GeoPandas CI.*

### CODE REVIEW

File: geopandas/tests/test_geoseries.py [hunk 1]
```diff
@@ -773,3 +773,158 @@ def test_explode_ignore_index(self):
         # index_parts is ignored if ignore_index=True
         s = s.explode(index_parts=True, ignore_index=True)
         assert_index_equal(s.index, expected_index)
+
+
+class TestGeoSeriesFromGeoJSON:
```
Comment (m-richards) [inline geopandas/tests/test_geoseries.py]: @Shikhar-Kesharwani do you think you could see about trying to align these new tests with the existing test suite structure? More specifically, that would mean adding test methods under TestConstructor here, or (the general preference) using toplevel test functions within a module, not embedding them in a class. We don't use autouse=True fixtures except in very specific circumstances.

Note also we don't insist on one assert per test, which can make tests more compact and readable.

Comment (Shikhar-Kesharwani) [reply geopandas/tests/test_geoseries.py]: Hi @m-richards, thank you for the feedback! I have updated the tests:
- Flattened into module-level top-level test functions (`test_from_geojson`, `test_from_geojson_crs`, `test_from_geojson_series`, `test_from_geojson_missing`, `test_from_geojson_on_invalid`) and removed the `autouse` fixture.
- Grouped related assertions to keep the tests concise and readable.

All tests pass cleanly.
